openshift/machine-config-operator
# Bug 1868158: gcp, azure: Handle azure vips similar to GCP

This PR does the following things:

- Rename gcp-routes-controller to apiserver-watcher, since it is generic
- Remove obsolete service-management mode from gcp-routes-controller
- Change downfile directory to /run/cloud-routes from /run/gcp-routes
- Write $VIP.up as well as $VIP.down
- Add an azure routes script that fixes hairpin.

Background: Azure hosts cannot hairpin back to themselves over a load balancer. Thus, we need to redirect traffic to the apiserver vip to ourselves via iptables. However, we should only do this when our local apiserver is running.

The apiserver-watcher drops a $VIP.up and $VIP.down file, accordingly, depending on the state of the apiserver. Then, we add or remove iptables rules that short-circuit the load balancer.

Unlike GCP, we don't need to do this for external traffic, only local clients.

**- How to verify it**
Install on azure, ensure connections to the internal API load balancer are reliable - both when the local apiserver process is running and stopped.

**- Description for the changelog**
Masters on azure can now reliably connect to the apiserver service, without encountering hairpin issues
